spec/models/batch_destroy_spec.rb

```diff
@@ -0,0 +1,82 @@
+require "spec_helper"
+
+describe BatchDestroy, type: :model do
+  let(:current_user) { create(:user, email: "current@user.com") }
+  let(:ability) { Ability.new(user: current_user, mission: get_mission) }
+  let(:destroyer) { BatchDestroy.new(batch, current_user, ability) }
+
+  describe "#destroy!" do
+    describe "user" do
+      let(:users) { create_list(:user, 3) }
+      let(:user) { create(:user, mission: create(:mission)) }
+      let(:batch) { users }
+
+      it "deletes everyone but the current user" do
+        users << current_user
+        destroyer.destroy!
+
+        # current user
+        expect(User.count).to eq(1)
+        expect(current_user.active).to be_truthy
```

Comment:
Shouldn't we be checking that `current_user` is not deleted? So I think `expect(current_user).not_to be_destroyed` should work. Or you could combine both of these expectations into `expect(User.all.to_a).to contain_exactly(current_user)`